File: compiler/src/main/java/com/aaroncoplan/waterfall/compiler/statements/ArrayIndexData.java
package com.aaroncoplan.waterfall.compiler.statements;

import com.aaroncoplan.waterfall.generated.WaterfallParser;

public class ArrayIndexData {

    public final String target;
    public final ExpressionData index;

    public ArrayIndexData(String filePath, WaterfallParser.ArrayIndexContext ctx) {
        this.target = ctx.target.getText();
        this.index = new ExpressionData(filePath, ctx.index);
    }
}

File: compiler/src/main/kotlin/com/aaroncoplan/waterfall/compiler/statements/ArrayIndexData.kt
package com.aaroncoplan.waterfall.compiler.statements

import com.aaroncoplan.waterfall.generated.WaterfallParser

class ArrayIndexData(filePath: String, ctx: WaterfallParser.ArrayIndexContext) {
    @JvmField val target: String = ctx.target.text
    @JvmField val index: ExpressionData = ExpressionData(filePath, ctx.index)
}
